pkg/api/validation/validation.go

```diff
@@ -1090,12 +1090,13 @@ func ValidateService(service *api.Service) errs.ValidationErrorList {
 		}
 	}
 
-	for _, ip := range service.Spec.DeprecatedPublicIPs {
+	for _, ip := range service.Spec.ExternalIPs {
 		if ip == "0.0.0.0" {
 			allErrs = append(allErrs, errs.NewFieldInvalid("spec.publicIPs", ip, "is not an IP address"))
 		} else if util.IsValidIPv4(ip) && net.ParseIP(ip).IsLoopback() {
 			allErrs = append(allErrs, errs.NewFieldInvalid("spec.publicIPs", ip, "publicIP cannot be a loopback"))
 		}
+		allErrs = append(allErrs, ValidateIpIsNotLinkLocalOrLoopback(ip)...)
```

Comment:
just 3 lines above this we check for loopback - that is redundant with this.  de-dup, please.